html/entities.ts

```diff
@@ -115,6 +129,52 @@ export function unescape(
     .replaceAll(RX_HEX_ENTITY, (_, hex) => codePointStrToChar(hex, 16));
 }
 
+/**
+ * A valid custom element name is a sequence of characters name that meets all of the following requirements:
+ *
+ * name must match the PotentialCustomElementName production:
+ *
+ * PotentialCustomElementName ::= [a-z] (PCENChar)* '-' (PCENChar)*
+ *
+ * PCENChar ::= "-" | "." | [0-9] | "_" | [a-z] | #xB7 | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x203F-#x2040] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
+ *
+ * This uses the EBNF notation from the XML specification. [XML]
+ *
+ * name must not be any of the following:
+ *
+ * annotation-xml
+ * color-profile
+ * font-face
+ * font-face-src
+ * font-face-uri
+ * font-face-format
+ * font-face-name
+ * missing-glyph
+ *
+ * @example Basic usage
+ *
+ * Using a valid custom element name
+ *
+ * ```ts
+ * import { isValidCustomElement } from '@std/html/entities'
+ * import { assertEquals } from "@std/assert";
+ *
+ * assertEquals(isValidCustomElement("custom-element"), true)
+ * ```
+ *
+ * Reference on invalid names can be found [here](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name)
+ *
+ * @param {string} elementName The element name to be validate
+ * @returns {boolean} A boolean value indicating if the custom element name is valid or not
```

Comment:
JSDoc `@param` types aren't need as we're using TypeScript.
```suggestion
 * @param elementName The element name to be validate
 * @returns A boolean value indicating if the custom element name is valid or not
```